src/freertos_drivers/esp_idf/EspIdfWiFi.hxx

```diff
@@ -0,0 +1,914 @@
+    {
+        // find an "empty" profie
+        OSMutexLock locker(&lock_);
+        int index = find_sta_profile("");
+        if (index >= 0)
+        {
+            str_populate<MAX_PASS_SIZE>(userCfg_.sta_[index].pass_, pass);
+            userCfg_.sta_[index].sec_ = sec_type_encode(sec_type);
+            str_populate<MAX_SSID_SIZE>(userCfg_.sta_[index].ssid_, ssid);
+            config_sync();
+        }
+        config_sync();
+        return index;
+    }
+
+    /// Delete a saved WiFi access point profile.
+    /// @param ssid access point SSID
+    /// @return profile index deleted, else -1 if profile not found
+    int profile_del(const char *ssid) override
+    {
+        int index = find_sta_profile(std::string(ssid));
+        if (index >= 0)
+        {
+            profile_del(index);
+        }
+        return index;
+    }
+
+    /// Delete a saved WiFi access point profile.
+    /// @param index index of profile to delete, 0xFF removes all
+    /// @return 0 if successful, else -1 if profile not found
+    int profile_del(uint8_t index) override
+    {
+        if (index >= HWDefs::MAX_STA_PROFILES)
+        {
+            return -1;
+        }
+        {
+            OSMutexLock locker(&lock_);
+            memset(userCfg_.sta_[index].ssid_, 0, MAX_SSID_SIZE);
+            memset(userCfg_.sta_[index].pass_, 0, MAX_PASS_SIZE);
+        }
+        config_sync();
+        return 0;
+    }
+
+    /// Get a saved WiFi access point profile by index.
+    /// @param index index within saved profilelist to get
+    /// @param ssid 33 byte array that will return the SSID of the index
+    /// @param sec_type will return the security type of the index
+    /// @param priority will return the priority of the index, may be unused
+    /// @return 0 upon success, -1 on error (index does not exist)
+    int profile_get(int index, char ssid[], SecurityType *sec_type,
+        uint8_t *priority) override
+    {
+        OSMutexLock locker(&lock_);
+        if (index >= HWDefs::MAX_STA_PROFILES)
+        {
+            return -1;
+        }
+        strncpy(ssid, userCfg_.sta_[index].ssid_, MAX_SSID_SIZE);
+        ssid[MAX_SSID_SIZE] = '\0';
+        *sec_type = sec_type_encode(userCfg_.sta_[index].sec_);
+        *priority = 0;
+        return 0;
+    }
+
+protected:
+    /// C structure version of the WiFi configuration settings. These are
+    /// essentially the connection profiles (SSID, password, security type).
+    template<size_t N> struct WiFiConfigNVS
+    {
+        uint32_t magic_; ///< magic number to detect initialization
+        WlanRole mode_; ///< role that the device will operate (AP, STA, etc.)
+        uint8_t padding_[3]; ///< padded for alignment
+        WiFiConfigCredentialsNVS ap_; ///< access point configuration
+        WiFiConfigCredentialsNVS sta_[N]; ///< station profiles
+    }; // template<size_t N> struct WiFiConfigNVS
+    static_assert(sizeof(WiFiConfigNVS<HWDefs::MAX_STA_PROFILES>::mode_) ==
```

Comment:
let's make a typedef for the actual wificonfignvs so that the template does not need to be written out everywhere:
template <size_t N> typedef WifiConfigNVSTemplate {...};
typedef WifiConfigNVSTemplate<HWDefs::MAX_STA_PROFILES> WifiConfigNVS;

Reply:
Done.